vms/platformvm/txs/mempool/mempool.go

```diff
@@ -64,24 +64,26 @@ type Mempool interface {
 	// possibly reissued.
 	MarkDropped(txID ids.ID, reason error)
 	GetDropReason(txID ids.ID) error
+
+	// Iterate iterates over the txs until f returns false
+	Iterate(f func(tx *txs.Tx) bool)
 }
 
 // Transactions from clients that have not yet been put into blocks and added to
 // consensus
 type mempool struct {
-	bytesAvailableMetric prometheus.Gauge
-	bytesAvailable       int
+	toEngine chan<- common.Message
 
-	unissuedTxs linkedhashmap.LinkedHashmap[ids.ID, *txs.Tx]
-	numTxs      prometheus.Gauge
+	bytesAvailableMetric prometheus.Gauge
+	numTxs               prometheus.Gauge
 
+	lock          sync.Mutex
```

Comment:
should we use a RWLock mutex to be able to RLock only when querying the mempool? Not sure it's be safe in case some of the attributes are updated while querying. Going to check